### PR TITLE
add authentication flow to json tabs

### PR DESCRIPTION
**Pull request by FaizChishtie**

### CODE REVIEW

File: src/components/AuthView.tsx [hunk 1]
```diff
@@ -1,20 +1,51 @@
...
 interface IAuthViewProps {
   title: string;
   url: string;
+  currentQuery: microsoftTeams.bot.QueryRequest;
+  onAuthenticated: (results: microsoftTeams.bot.Results) => void;
 }
 
 export const AuthView: React.FC<IAuthViewProps> = (props: IAuthViewProps): JSX.Element => {
+  const onAuth = (results: microsoftTeams.bot.Results): void => {
+    alert('Successfully authenticated.');
```
Comment (sachingupta) [inline src/components/AuthView.tsx]: remove alert

File: src/components/AuthView.tsx [hunk 2]
```diff
@@ -1,20 +1,51 @@
...
+    props.onAuthenticated(results);
+  };
+
+  const onError = (error: string): void => {
+    alert(`Failed to authenticate.\n${error}`);
```
Comment (sachingupta) [inline src/components/AuthView.tsx]: use error view

File: src/components/AuthView.tsx [hunk 3]
```diff
@@ -1,20 +1,51 @@
...
 
 interface IAuthViewProps {
   title: string;
   url: string;
+  currentQuery: microsoftTeams.bot.QueryRequest;
+  onAuthenticated: (results: microsoftTeams.bot.Results) => void;
 }
 
 export const AuthView: React.FC<IAuthViewProps> = (props: IAuthViewProps): JSX.Element => {
+  const onAuth = (results: microsoftTeams.bot.Results): void => {
```
Comment (sachingupta) [inline src/components/AuthView.tsx]: may be onAuthSuccess

File: src/components/AuthView.tsx [hunk 4]
```diff
@@ -1,20 +1,51 @@
...
+    alert('Successfully authenticated.');
+    props.onAuthenticated(results);
+  };
+
+  const onError = (error: string): void => {
```
Comment (sachingupta) [inline src/components/AuthView.tsx]: may be onAuthFail

File: src/components/AuthView.tsx [hunk 5]
```diff
@@ -1,20 +1,51 @@
...
+    alert(`Failed to authenticate.\n${error}`);
+  };
+
+  React.useEffect((): void => {
```
Comment (sachingupta) [inline src/components/AuthView.tsx]: this is not needed here as these are need for 1st time when tab load

File: src/components/AuthView.tsx [hunk 6]
```diff
@@ -1,20 +1,51 @@
...
       <Text size={'large'} content={props.title} />
       <Text
         size={'medium'}
         content={
           <p>
-            You&apos;ll need to <a href={props.url}>sign in</a> to use this app.
+            You&apos;ll need to{' '}
+            <a href="" onClick={handleAuthentication}>
+              sign in
+            </a>{' '}
```
Comment (sachingupta) [inline src/components/AuthView.tsx]: use css and <a> tag to give some space on right

File: src/components/AuthView.tsx (hunk 2 above)
Comment (FaizChishtie) [reply src/components/AuthView.tsx]: I think we shouldn't default to that - i'll write a message underneath like compose extenison 

errorsn

File: src/components/ContentView.tsx [hunk 7]
```diff
@@ -69,6 +67,11 @@ export const ContentView: React.FC<IContentViewProps> = (props: IContentViewProp
     }
   };
 
+  const handleAuthenticated = (results: microsoftTeams.bot.Results) => {
```
Comment (sachingupta) [inline src/components/ContentView.tsx]: try to refactor onResults and handleAuthenticated function.

**Pull request merged**